### ISSUE TITLE
Update Error: Ice can’t be updated if it’s running from the location it was downloaded to.

### ISSUE THREAD

**Issue by AeliaJBL** [jordanbaird/Ice#121]
### Check existing issues

- [X] I have checked existing issues and believe that my issue is not a duplicate

### Description

When attempting to manually check for updates, I receive the following error: "Ice can’t be updated if it’s running from the location it was downloaded to." I have uninstalled and reinstalled, but the same behavior appears.

### Steps to Reproduce

Open the app settings and click on "check for updates."

### Ice Version

0.9.0

### macOS Version

14.5

**Comment by jordanbaird** [comment]
Do you have Ice in your "Applications" folder?

**Comment by jordanbaird** [comment]
You're welcome!

**Comment by dankilman** [comment]
Hi!
I see this ticket was closed just now but it makes sense to post here as I'm basically getting the same thing but I have verified and even redownloaded Ice and moved it to `/Applications` again to verify.
So I'm getting this and it is in `/Applications`. My only thought is perhaps that the permissions for Ice.app are my user's permissions and not `root:wheel` as most apps that are installed via automated scripts

**Comment by jordanbaird** [comment]
@dankilman Did you move Ice to `/Applications` using Finder? macOS imposes a quarantine on the app bundle if you don't do this. You can clear it by either redownloading and using Finder to move the app, or by running `xattr -dr com.apple.quarantine /Applications/Ice.app` in Terminal.

**Comment by dankilman** [comment]
Perfect! Moving via Finder did the trick. Many thanks 🙏

**Comment by TheRealRino** [comment]
Hi, this issue still shows up when in the downloaded folder there's no Ice.app. I'm using version 0.10.5

**Comment by jordanbaird** [comment]
@TheRealRino Did you drag it using Finder? Not using a third party viewer at any point in between? Also, did you try running `xattr -dr com.apple.quarantine /Applications/Ice.app`?

**Comment by TheRealRino** [comment]
yes I am only using Finder and not another third party viewer. Also I tried running that xattr -dr com.apple.quarantine /Applications/Ice.app in Terminal but says "Operation not permitted"

**Comment by jordanbaird** [comment]
@TheRealRino Try `sudo xattr -dr com.apple.quarantine /Applications/Ice.app`. You'll have enter your password.

**Comment by TheRealRino** [comment]
okay I had tried allowed Full Disk Access to the "Terminal App" and input the "xattr -dr com.apple.quarantine /Applications/Ice.app" but it does nothing. The error still pops up. Or is it the error pops up becuase I am already using the latest version? I also did update Ice using "CleanMyMacX App".

**Comment by jordanbaird** [comment]
@TheRealRino What does the error say specifically? Could you screenshot it?

It very well could have been caused by CleanMyMac. Even though you originally used Finder to install Ice, the original app bundle will have been replaced when you updated this way. I would recommend not using CleanMyMac to update apps, as it could cause this same error for other apps as well. I think uninstalling Ice and reinstalling using Finder is your best bet here.

**Comment by TheRealRino** [comment]
The same error in this thread. "Ice can’t be updated if it’s running from the location it was downloaded to."

Anyway, thank you for letting me know. I reinstalled the app and it works.
